api/pdf.py

```
@@ -168,9 +170,12 @@ def footer(self):
 
     except Exception as e:
-        print(f"Failed to generate PDF: {e}")
-        import traceback
-        traceback.print_exc()
+        _logger.error(
+            "Failed to generate PDF: %s (%s)",
+            e,
+            type(e).__name__,
+            exc_info=True,
+        )
         return jsonify({"error": f"Failed to generate PDF: {str(e)}"}), 500
```

Ruff (0.15.14)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)